lib/pry/config/behavior.rb

```diff
@@ -1,6 +1,41 @@
 class Pry
   class Config < Pry::BasicObject
-    # rubocop:disable Metrics/ModuleLength
+    #
+    # {Pry::Config::Behavior} is a module who can be included by classes who
+    # wish to behave similar to an OpenStruct object:
```

Comment:
The docs don't explain how this is different to OpenStruct, which makes me wondering why not to use OpenStruct instead.

Reply:
It can explains how it is different just below this paragraph.

__1.__

```
    # Classes who include {Pry::Config::Behavior} can be linked to each other
    # to provide a default in case a key does not exist locally:
```

__.2__

```
    # When an object is read from a default like in the example above, a copy
    # of the object is created to avoid a mutation changing its value:
```

__3.__

It is not mentioned here  in the docs because it is covered in other parts, but OpenStruct does not support `Pry.lazy {}` while `Pry::Config::Behavior` does.

Reply:
__4.__

Not documented here but worth listing: `Pry::Config::Behavior` supports eager and lazy loading, allowing Pry to not have a hard dependency on 'Readline' (for example).

Reply:
> But your presented behaviour shows that it is nothing like OpenStruct, so why to mention it at all?

The first example shows that the interface is exactly like OpenStruct, for reading and setting of keys. The examples after that demonstrate behaviour that is unique to `Behaviour`, but to say it is "nothing like OpenStruct" is inaccurate. 

> It's not obvious from the docs at all. I expect to see something like "Unlike OpenStruct, this can A, B, C".

Okay, i can probably reword it to be something like this.